### PR TITLE
TINY-2827 Update textpattern docs with TINY-2991 changes

### PR DESCRIPTION
**Pull request by blackcathikari**

### CODE REVIEW

**Comment by shikhanansi** [comment]
@blackcathikari please review the changes one more time. This PR can be merged if you're happy with the changes.

File: plugins/textpattern.md [hunk 1]
```diff
@@ -53,7 +53,12 @@ There are three types of patterns: inline, block, and replacement patterns. Inli
 
 #### Inline patterns
 
-Inline patterns have a `start` and an `end`, and either a `format` or a `cmd` and optional `value` property. This allows for patterns to be used to either apply a format or execute a command, optionally with the given value.
+Inline patterns must have the following:
+* A `start` and an `end`
+* A `format` or a `cmd` 
+* An optional `value` property.
```
Comment (tiny-james) [inline plugins/textpattern.md]: The `value` property is only allowed when `cmd` is specified.

File: plugins/textpattern.md [hunk 2]
```diff
@@ -71,13 +76,21 @@ tinymce.init({
...
+* `{start: '**', end: '**', format: 'bold'}` - Entering text between`**` and then pressing the **spacebar** will result in the `bold` format being applied. 
+* `{start: '~', end: '~', cmd: 'createLink', value: 'https://tiny.cloud'}` - This executes `editor.execCommand('createLink', false, 'https://tiny.cloud')`, which will wrap the text between the `~` symbols in an link that points to `https://tiny.cloud`.
 
 #### Block patterns
 
-Block patterns have a `start` and either a `format` or a `cmd` and optional `value` property. They do not have an `end` property. This allows for patterns to be used to either apply a block format or execute a command, optionally with the given value.
+Block patterns must have the following:
+* A `start` 
+* A `format` or a `cmd` 
+* An optional `value` property.
```
Comment (tiny-james) [inline plugins/textpattern.md]: Again the `value` property only works with `cmd`.

**Comment by blackcathikari** [comment]
@shikhanansi It's just the parts James commented on. I purposefully didn't put the properties in lists because of how complicated it is to be honest, but my sentence version wasn't particularly clear either. E.g. for block patterns you must have a `start`, but then you either have a `format` or you have a `cmd` with an optional `value`.

File: plugins/textpattern.md [hunk 3]
```diff
@@ -71,13 +76,21 @@ tinymce.init({
 });
 ```
 
-Using the configuration in this example, typing `*`, some text, `*`, then space will result in the format `italic` being applied to the text between the `*` symbols. The same will happen for `**` and `bold`. The third pattern will execute `editor.execCommand('createLink', false, 'https://tiny.cloud')`, which will wrap the text between the `~` symbols in an link that points to `https://tiny.cloud`.
+Using the configuration in this example:
+* `{start: '*', end: '*', format: 'italic'}` - Entering text between`*` and then pressing the **spacebar** will result in the `italic` format being applied to the text between the `*` symbols. 
+* `{start: '**', end: '**', format: 'bold'}` - Entering text between`**` and then pressing the **spacebar** will result in the `bold` format being applied. 
+* `{start: '~', end: '~', cmd: 'createLink', value: 'https://tiny.cloud'}` - This executes `editor.execCommand('createLink', false, 'https://tiny.cloud')`, which will wrap the text between the `~` symbols in an link that points to `https://tiny.cloud`.
```
Comment (blackcathikari) [inline plugins/textpattern.md]: "an link"

File: plugins/textpattern.md [hunk 4]
```diff
@@ -129,4 +148,8 @@ tinymce.init({
 });
 ```
 
-In this example, typing `---` will insert a horizontal rule block, but typing `(c)` will insert an inline copyright symbol. This is useful for commonly used phrases or symbols, and can be leveraged to create content templates. The last pattern is an example of this.
+Using the configuration in this example:
+* Typing `---` will insert a horizontal rule block.
```
Comment (blackcathikari) [inline plugins/textpattern.md]: This should possibly say "Typing x then space or enter"?

File: plugins/textpattern.md (hunk 1 above)
Comment (shikhanansi) [reply plugins/textpattern.md]: Fixed

File: plugins/textpattern.md (hunk 2 above)
Comment (shikhanansi) [reply plugins/textpattern.md]: Fixed

File: plugins/textpattern.md (hunk 3 above)
Comment (shikhanansi) [reply plugins/textpattern.md]: Fixed

File: plugins/textpattern.md (hunk 4 above)
Comment (shikhanansi) [reply plugins/textpattern.md]: Fixed

**Comment by shikhanansi** [comment]
Looks good to me. @blackcathikari this PR can be merged.
Regards,
Shikha

**Comment by blackcathikari** [comment]
@shikhanansi Looks like the content got duplicated, from the first example to the end? I noticed it because the Options heading appears twice...

**Comment by shikhanansi** [comment]
@blackcathikari please review and merge the PR

**Comment by shikhanansi** [comment]
@lnewson could you please stamp this PR. We need this content for the release tomorrow.

**Comment by lnewson** [comment]
@shikhanansi I don't know this well enough sorry... I'll ask if someone else can take a look who knows it better.

**Pull request merged**